packages/remix/src/generators/application/files/remix.config.js__tmpl__

```diff
@@ -7,4 +7,5 @@ module.exports = {
   // assetsBuildDirectory: "public/build",
   // serverBuildPath: "build/index.js",
   // publicPath: "/build/",
+  watchPaths: ["<%= offsetFromRoot %>libs"]
```

Comment:
Rather than including all of the libs into the watchPaths, should we potentially use the `createGlobPatternsForDependencies` function? [link](https://github.com/nrwl/nx/blob/1769edb2ed944cdd58372b846c0c461df74606b0/packages/workspace/src/utilities/generate-globs.ts#L15)

We use it in the Angular and [React](https://github.com/nrwl/nx/blob/1769edb2ed944cdd58372b846c0c461df74606b0/packages/react/tailwind.ts#L8) when generating the purge globs for Tailwind: https://github.com/nrwl/nx/blob/1769edb2ed944cdd58372b846c0c461df74606b0/packages/react/src/generators/setup-tailwind/files/tailwind.config.js__tmpl__#L11

(we can do that in a separate PR to not bloat this one too much)

Reply:
I think that would have a negative impact on DX. `watchPaths` is only read at start-up, as far as I know. If we read the deps at start-up time, we wouldn't be watching for any new deps that are added to the application. It would build fine, but wouldn't hot-reload when changes were made down in the newly dependent library.

One thing we might experiment with is instead of watching the entire libs directory, we could grab the `sourceRoot` of all libraries and use that instead. I'm not sure what's more performant for Remix: a single large directory with a lot of files or a lot of directories with a smaller number of files. It's worth some investigation, but I would agree that it should be a separate PR.

Reply:
> watchPaths is only read at start-up, as far as I know. If we read the deps at start-up time, we wouldn't be watching for any new deps that are added to the application. 

Yeah I know. That's the downside.. I think there are two things
- perf impact of monitoring all libs. We'd need to get some real measurements here to see the impact. I'd suspect that most folks wouldn't notice the difference, but in a large monorepo you totally might
- whether you'd want to reload the Remix compilation for every lib change which might even be completely unrelated because the lib is not even used by the Remix app in the larger monorepo


> One thing we might experiment with is instead of watching the entire libs directory, we could grab the sourceRoot of all libraries and use that instead. I'm not sure what's more performant for Remix

I don't think that has any noticeable impact. I'd rather just go for the entire libs folder or selectively pull deps in.

Reply:
I will leave this as-is for now, and follow up with investigation on performance later.